core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -118,9 +118,11 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    * Can be generated using [[org.apache.spark.scheduler.InputFormatInfo.computePreferredLocations]]
    * from a list of input files or InputFormats for the application.
    */
+  @Deprecated("Passing in preferred locations has no effect at all, see SPARK-8949", "1.5.0")
```

Comment:
This is the `java.lang` annotation. Does it work with Scala? I thought we used `@deprecated` for this -- better for consistency even if both work